Amqp/src/Akka.Streams.Amqp/AmqpRpcFlowStage.cs

```diff
@@ -129,22 +139,42 @@ public override void WhenConnected()
                 Pull(_stage.In);
 
                 // we have only one consumer per connection so global is ok
-                Channel.BasicQos(0, (ushort)_stage.BufferSize, false);
-                var consumerCallback = GetAsyncCallback<IncomingMessage>(HandleDelivery);
-                _amqpSourceConsumer = new DefaultConsumer(consumerCallback, shutdownCallback);
+                Channel.BasicQos(0, (ushort) _stage.BufferSize, false);
+
+                var consumerCallback = GetAsyncCallback<CommittableIncomingMessage>(HandleDelivery);
+
+                var commitCallback = GetAsyncCallback<ICommitCallback>(callback =>
+                {
+                    switch (callback)
+                    {
+                        case AckArguments args:
+                            Channel.BasicAck(args.DeliveryTag, args.Multiple);
+                            if (--_unackedMessages == 0) CompleteStage();
+                            args.Promise.SetResult(Done.Instance);
```

Comment:
`SetResult` is potentially risky (it may throw). If we never use set exception or set cancelled, it's ok. Otherwise, we probably should change that to `TrySetResult`.

Reply:
Done.